code/datums/mob_hud.dm

```diff
@@ -268,15 +268,21 @@ var/datum/mob_hud/huds = list(
 		var/datum/limb/head = get_limb("head")
 		var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
 		var/revive_enabled = 1
+		var/stage = 1
 		if(world.time - timeofdeath > revive_grace_period || undefibbable)
 			revive_enabled = 0
-		else
-			if(suiciding || !head || !head.is_usable() || !heart || heart.is_broken() || !has_brain() || chestburst || (HUSK in mutations) || !mind)
+		else if(suiciding || !head || !head.is_usable() || !heart || heart.is_broken() || !has_brain() || chestburst || (HUSK in mutations) || !mind)
+			revive_enabled = 0
+		else if(!client)
+			var/mob/dead/observer/G = get_ghost()
+			if(!istype(G))
 				revive_enabled = 0
-			else if(!client)
-				var/mob/dead/observer/G = get_ghost()
-				if(!istype(G))
-					revive_enabled = 0
+
+		if(stat == DEAD && !undefibbable)
+			if((world.time - timeofdeath) > ((revive_grace_period / 4) * 2))
+				stage = 2
+			else if((world.time - timeofdeath) > ((revive_grace_period / 4) * 3))
```

Comment:
```suggestion
			else if((world.time - timeofdeath) > (revive_grace_period * 0.75))
```
I feel this is much more clear, but feel free to do a `* (3/4)` if you prefer.

Reply:
Since `revive_grace_period` is 5 minutes, you could also give 2 minutes for the first stage, thus doing `* 0.4` in the first check, and plus another 2 minutes for the second, doing `* 0.8`. A full minute to the last stage seems fair, and a proper indicator of urgency.